### ISSUE TITLE
wiki repositories don't refresh because they're not part of the API

### ISSUE THREAD

**Issue by shiftkey** [desktop/desktop#2061]
### Description

While we can now clone wiki repositories in Desktop, there are some other corner cases (and console errors) that light up as a result of this.

### Version

**GitHub Desktop version:** `master` - e.g. 77c2146e4094ed50904ecb0c378f245a000e6686
**OS version:** macOS Sierra 10.12.5

### Steps to Reproduce

- Launch the app
- Add an existing GitHub wiki repository to the app via drag and drop
- Confirm it appears as a GitHub repository

<img width="978" src="https://user-images.githubusercontent.com/359239/27323592-554fad9a-55a2-11e7-9a80-f0a682cdaaa6.png">

- Peep the dev console, see the errors for all the nonexistent endpoints:
   - fetching issues
   - fetching mentionable users
   - fetching repository details
   - fetching the refresh interval from `/repos/{owner}/{name}.wiki/git`

<img width="1436"  src="https://user-images.githubusercontent.com/359239/27323654-9201f5b8-55a2-11e7-9ca2-1ecacbba76f6.png">

This is using https://github.com/aspnet/Home/wiki

**Expected behaviour:** avoid fetching these endpoints for known cases
**Actual behaviour:** lots of errors

**Reproduces how often:** I only tested this a couple of times, but I'm pretty sure it's 100%

**Comment by joshaber** [comment]
Yeah, that's a good shout. I knew this was a problem but didn't document it.

**Comment by shiftkey** [comment]
The "View on GitHub" link also doesn't detect wiki repositories. Whatever solution we settle on here, the link shouldn't render for a wiki because there's no UI to display on GitHub or GitHub Enterprise for it.

**Comment by j-f1** [comment]
@shiftkey

```
https://github.com/:user/:repo/wiki/:page_name/_compare/:sha
```

**Comment by shiftkey** [comment]
@j-f1 nice find - it's a slightly different resource format (the repository itself is hosted at is `https://{host}/:user/:repo.wiki/` but all the website resources are at `https://{host}/:user/:repo/wiki` and we'd need to look at the commit contents itself to get `:page_name`).

I think once we figure out how to distinguish wiki repositories from regular repositories, we can glue together the other sorts of features like that...

**Comment by mischnic** [comment]
Just wanted to add that wikis belonging to private repositories have a public icon.

**Comment by shiftkey** [comment]
Related to this change, #4092 reports that wiki repositories are categorized as "Other"

![](https://user-images.githubusercontent.com/177608/36598887-ffbd628c-1883-11e8-8326-b62f1ab61530.png)

**Comment by joaomlneto** [comment]
Anyone working on this? Deferred until the GitHub API supports wiki repositories? :-) I'm feeling a bit adventurous.

**Comment by shiftkey** [comment]
@joaomlneto nope, but feel free to have a go- I'm happy to answer any questions about things you stumble upon!

**Comment by Andre601** [comment]
I can confirm that fetching latest changes (which shouldn't anything other than fetching from any git-repo) doesn't do anything.
GitHub Desktop just stays silent and doesn't react at all.

Can this please be fixed? It's annoying to delete the wiki folder and download/clone it again, just to have the latest changes of the wiki...
